fili-core/src/test/groovy/com/yahoo/bard/webservice/web/DataApiRequestSortSpec.groovy

```diff
@@ -0,0 +1,142 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web
+
+import com.yahoo.bard.webservice.application.JerseyTestBinder
+import com.yahoo.bard.webservice.druid.model.orderby.OrderByColumn
+import com.yahoo.bard.webservice.druid.model.orderby.SortDirection
+import com.yahoo.bard.webservice.web.endpoints.DataServlet
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class DataApiRequestSortSpec extends Specification {
+
+    JerseyTestBinder jtb
+
+    def setup() {
+        // Create the test web container to test the resources
+        jtb = new JerseyTestBinder(DataServlet.class)
+    }
+
+    def cleanup() {
+        // Release the test web container
+        jtb.tearDown()
+    }
+
+
+
+    def "If dateTime is not the first value in the sort list, then throw an error"() {
+        setup:
+        String expectedMessage = ErrorMessageFormat.DATE_TIME_SORT_VALUE_INVALID.format()
+
+        when:
+        new DataApiRequest().generateDateTimeSortColumn(["xyz":SortDirection.DESC,"dateTime":SortDirection.DESC])
+
+        then:
+        Exception e = thrown(BadApiRequestException)
+        e.getMessage() == expectedMessage
+    }
+
+    @Unroll
+    def "Validate the sort column and direction map from #sortString string"() {
+        when:
+        Map<String, SortDirection> actual = new DataApiRequest().generateSortColumns(sortString)
+
+        then:
+        actual == expected
+
+        where:
+        sortString                        | expected
+        "dateTime"                        | ["dateTime":SortDirection.DESC] as Map
+        "dateTime|ASC"                    | ["dateTime":SortDirection.ASC] as Map
+        "dateTimexyz|DESC"                | ["dateTimexyz":SortDirection.DESC] as Map
+        "xyz,dateTime,abc"                | ["xyz":SortDirection.DESC,"dateTime":SortDirection.DESC, "abc":SortDirection.DESC] as Map
+        "xyz,dateTime|DESC,abc"           | ["xyz":SortDirection.DESC,"dateTime":SortDirection.DESC, "abc":SortDirection.DESC] as Map
+        "xyz|DESC,dateTime|DESC,abc|ASC"  | ["xyz":SortDirection.DESC,"dateTime":SortDirection.DESC, "abc":SortDirection.ASC] as Map
+        "xyz|DESC,dateTime,abc|DESC"      | ["xyz":SortDirection.DESC,"dateTime":SortDirection.DESC, "abc":SortDirection.DESC] as Map
+        "xyz|DESC,dateTime"               | ["xyz":SortDirection.DESC,"dateTime":SortDirection.DESC] as Map
+        ""                                | null
+        "dinga"                           | ["dinga":SortDirection.DESC] as Map
+
+    }
+
+    @Unroll
+    def "Generate dateTime sort column from columnDirection map "() {
```

Comment:
Should have some sort of expanded variable in the test name when we have an `@Unroll` test, to differentiate the cases.